src/datastore/postgres/PgDatastore.ts

```diff
@@ -292,6 +292,69 @@ export class PgDatastore implements Datastore {
         await this.postgresDb.none(statement, {matrixuser, roomid});
     }
 
+    public async upsertUserMetrics(matrixId: string, remote: boolean, puppeted: boolean): Promise<void> {
+        return;
+    }
+
+    public async upsertRoomMetrics(roomId: string, type: RoomType): Promise<void> {
+        return;
+    }
+
+    public async upsertActivityMetrics(userId: string, room: BridgedRoom, date?: Date): Promise<void> {
+        date = date || new Date();
+
+        const roomId = `${room.MatrixRoomId}|${room.SlackChannelId}`;
+        await this.postgresDb.none("INSERT INTO metrics_user_room_activities VALUES(${userId}, ${roomId}, ${date})", {
+            userId,
+            roomId,
+            date: `${date.getFullYear()}-${date.getMonth()}-${date.getDate()}`,
+        });
+        return;
+    }
+
+    public async getActiveRoomsPerTeam(activityThreshholdInDays = 2, historyLengthInDays = 30): Promise<any> {
+        return (await this.postgresDb.manyOrNone(
+            "SELECT team_id, room_id, COUNT(*) AS active_days" +
+            "FROM metrics_user_room_activities" +
+            "WHERE date >= DATE_SUB(CURRENT_DATE, INTERVAL ${historyLengthInDays} DAYS)" +
+            "INNER JOIN metrics_rooms ON metrics_user_room_activities.room_id = metrics_rooms.room_id" +
+            "GROUP BY team_id, room_id" +
+            "HAVING active_days > ${activityThreshholdInDays}",
+            { activityThreshholdInDays, historyLengthInDays },
+        )).map((u) => ({
+            teamId: u.matrixuser,
+            activeRooms: u.activeRooms,
+        }));
+        // return {
+        //     "ABCDEFGH": [
+        //         "#general:localhost|ABCDEFG",
+        //         "#random:localhost|BCDEFGH",
+        //         "#finances:localhost|CDEFGHI",
+        //     ]
+        // };
+    }
+
+    public async getActiveUsersPerTeam(activityThreshholdInDays = 2, historyLengthInDays = 30): Promise<any> {
+        return (await this.postgresDb.manyOrNone(
+            "SELECT team_id, user_id, COUNT(*) AS active_days" +
+            "FROM metrics_user_room_activities" +
+            "WHERE date >= DATE_SUB(CURRENT_DATE, INTERVAL ${historyLengthInDays} DAYS)" +
+            "INNER JOIN metrics_users ON metrics_user_room_activities.user_id = metrics_users.user_id" +
+            "GROUP BY team_id, user_id" +
```

Comment:
Hmm, we just want the count of users mapped against the team_id, rather than the user_ids themselves. Although, we can just count it so it's probably not important